src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -737,15 +740,48 @@ event::ptr primitive_inst::realloc_if_needed() {
 
     // Clear out memory if was previously reused, but now primitive can't be optimized
     if (!_node->is_type<concatenation>() && (_node->is_runtime_skippable() || _node->is_type<crop>())) {
+        std::function<void(cldnn::primitive_inst*, cldnn::memory::ptr)> reset_user_output_memory;
+        reset_user_output_memory = [&](cldnn::primitive_inst* curr_inst, cldnn::memory::ptr input_mem_ptr) {
+            auto curr_output_memory_ptr = curr_inst->output_memory_ptr(0);
+            if (curr_inst->can_be_optimized()
+                    && curr_inst->get_node().is_runtime_skippable()
```

Comment:
Um, Now I think, this should be applied even though the user is permanently opt out, (not runtime skippble) becuase if it is so, it will always try to update memory with its parent.

Reply:
Fixed it